### PR TITLE
Add codespell configuration and apply spelling changes

### PR DESCRIPTION
**Pull request by namurphy**
[Codespell](https://github.com/codespell-project/codespell) is a spellcheck tool specifically intended for source code.  Instead of looking for words not found in a dictionary, it looks for words that match common misspellings.  Consequently, codespell leads to very few false positives, but will miss some misspelled words that are not contained in its dictionaries for misspellings.  Overall, it's a great tool that we've been using over in PlasamPy and I highly recommend it!  

This PR adds a configuration for codespell in `setup.cfg` which specifies which files to ignore and which false positives to allow.  It also applies the spelling changes found by codespell.  I'll comment on a few of these which I'm not sure about.  I'm planning to add a pre-commit hook for codespell.

### CODE REVIEW

File: docs/whatsnew/0.9.rst [hunk 1]
```diff
@@ -36,7 +36,7 @@ By the numbers:
 * 19 new contributors
 
 There have been numerous improvements to large parts of SunPy, notably in the content of SunPy's documentation, and the continuous integration testing of SunPy.
-In addition, SunPy co-ordinate system functionality has been improved, and the transformation between SunPy specific co-ordinate systems and those implemented by Astropy is now tested.
+In addition, SunPy coordinate system functionality has been improved, and the transformation between SunPy specific coordinate systems and those implemented by Astropy is now tested.
```
Comment (namurphy) [inline docs/whatsnew/0.9.rst:39]: I looked it up and "co-ordinate" is an acceptable alternate spelling, though leaving the hyphen out makes the spelling more consistent with `sunpy.coordinates`.

**Comment by nabobalis** [comment]
Can we add it to the pre-commit as well?

**Comment by namurphy** [comment]
> Can we add it to the pre-commit as well?

Hopefully!  For some reason, when I do `pre-commit run --all` with the hook I'm currently trying, it's not picking up the configuration in `setup.cfg`.  For PlasmaPy, we have a tox environment for codespell that gets run via a GitHub Action, which is a backup possibility.

**Comment by namurphy** [comment]
> You might need to specify the config file in the pre-commit. I think we do that for some of the other hooks.

Good idea!  Looks like that is working.

Also, it looks like the changelog check is failing because this PR is editing a changelog file for a separate pull request.  Adding the "No Changelog Entry Needed" label should fix that.

**Review by ConorMacBride — APPROVED** [review]
This is brilliant! So many fixes 😬

**Review by Cadair — APPROVED** [review]
This is great thanks a lot @namurphy :rocket:

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -69,5 +69,10 @@ repos:
         pass_filenames: false
         always_run: true
 
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.2.2
+    hooks:
+      - id: codespell
+        args: ['--config setup.cfg']
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply .pre-commit-config.yaml:76]: I am pretty sure (90%), that it doesnt change anything, it just tells you.

**Comment by dstansby** [comment]
❤️ thanks so much for adding this @namurphy , has already caught a few of my spelling mistakes in new PRs!

**Comment by Cadair** [comment]
Yeah this is great :D

**Pull request merged**